substrate/frame/revive/src/lib.rs

```diff
@@ -1587,72 +1644,84 @@ impl<T: Config> Pallet<T> {
 	pub fn bare_instantiate(
 		origin: OriginFor<T>,
 		evm_value: U256,
-		gas_limit: Weight,
-		mut storage_deposit_limit: BalanceOf<T>,
+		transaction_limits: TransactionLimits<T>,
 		code: Code,
 		data: Vec<u8>,
 		salt: Option<[u8; 32]>,
 		exec_config: ExecConfig<T>,
 	) -> ContractResult<InstantiateReturnValue, BalanceOf<T>> {
-		let mut gas_meter = GasMeter::new(gas_limit);
-		let mut storage_deposit = Default::default();
+		let mut transaction_meter = match TransactionMeter::new(transaction_limits) {
+			Ok(transaction_meter) => transaction_meter,
+			Err(error) => return ContractResult { result: Err(error), ..Default::default() },
+		};
+
 		let try_instantiate = || {
 			let instantiate_account = T::InstantiateOrigin::ensure_origin(origin.clone())?;
 
 			if_tracing(|t| t.instantiate_code(&code, salt.as_ref()));
-			let (executable, upload_deposit) = match code {
+			let executable = match code {
```

Comment:
Now need to do any calculation of the `upload_deposit` in this function. This is now correctly done with the `transaction_meter` created before calling `try_upload_code`.

Reply:
should we stop returning an enum here then?
let (executable, ..) = Self::try_upload_code(

Reply:
nvm then if it makes things more complicated, I was under the impression that in the rare case where it was used 
we could just read the deposit from the updated meter reference

Reply:
That's true, I simplified according to your suggestion in https://github.com/paritytech/polkadot-sdk/pull/10166/commits/2dba778d50469029045d226ec73fe7a2dd81fc57